salt/cloud/clouds/ec2.py

```diff
@@ -4205,7 +4205,7 @@ def show_keypair(kwargs=None, call=None):
         return False
 
     params = {'Action': 'DescribeKeyPairs',
-              'KeyName.1': kwargs['keyname']}
```

Comment:
@techhat You're right. I didn't read that documentation. I thought (a little bit too fast) that it would have the same behaviour.
It is fixed in a new commit.
Thanks for the review